### PR TITLE
FILE-13 - Add more test coverage

### PR DESCRIPTION
**Pull request by gitstart**
### What does this PR do?

- Add more unit test coverage
- Add more e2e test coverage 

### Issue reference

https://github.com/filecoin-saturn/L1-dashboard/issues/50

### Demo video/scrennshot:

**Added unit tests for each component**
image

**Added E2E test for app**
image

**Added E2E test for WebUi mode**

image

###  Demo E2E
https://www.loom.com/share/c1fa9d4d5cfd418badf6b4f0e93ad38d

### Coverage snapshot
image
image

> This code was written and reviewed by GitStart Community. Growing future engineers, one PR at a time.

### CODE REVIEW

**Comment by gitstart** [comment]
#### `env` for `webui` 

- VITE_ROOT_DOMAIN
- VITE_METRICS_ORIGIN
- VITE_STATS_ORIGIN
- VITE_MODE_WEBUI

#### `env` for `embed` mode

- VITE_ROOT_DOMAIN
- VITE_METRICS_ORIGIN
- VITE_STATS_ORIGIN

**Comment by gitstart** [comment]
@guanzo this PR is ready for review

File: src/pages/stats/index.tsx [hunk 1]
```diff
@@ -52,9 +52,9 @@ export default function Stats() {
   }
 
   return (
-    <>
+    <div data-testid="stats-grid" className="h-full">
```
Comment (juliangruber) [inline src/pages/stats/index.tsx:55]: Is adding `.h-full` related to tests?

**Review by juliangruber — CHANGES_REQUESTED** [review]
These new tests are great!

I'm just wondering about the stability of this project - will adding tests with this level of detail slow down the project, or speed it up?

File: src/pages/stats/index.tsx (hunk 1 above)
Comment (juliangruber) [reply src/pages/stats/index.tsx:55]: nvm I see this is because it used to be a fragment. I trust this will render exactly the same?

**Review by DiegoRBaquero — COMMENTED** [review]
Not really familiar with FE or its testing, but, a lot of these assertions seem to be just asserting what the code is telling it to be. If we define button to have a width of X why would it be Y on the test?

File: src/pages/stats/index.tsx (hunk 1 above)
Comment (gitstart) [reply src/pages/stats/index.tsx:55]: Yes this would not distort the view at all

**Comment by gitstart** [comment]
> Not really familiar with FE or its testing, but, a lot of these assertions seem to be just asserting what the code is telling it to be. If we define button to have a width of X why would it be Y on the test?

Thank you for the question. The test is of two kinds, an e2e test and a unit test The e2e test uses Puppeteer to ensure that the features work from end to end on the browser and no new additions break existing features. Same as the unit test. It ensures that it renders as expected with the appropriate parameters and that future additions would not cause breaking changes.

**Pull request closed without merging**